File: 128-longest-consecutive-sequence/longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        if(nums == null || nums.length == 0){
            return 0;
        }

        Map<Integer, Boolean> visited = new HashMap<>();
        for(int num: nums){
            visited.put(num, false);
        }
        int longest = 0;

        for(int num: nums){
            if(visited.get(num)){
                continue;
            }
            visited.put(num, true);

            int continueStreak = 1;

            for(int i=num+1; visited.containsKey(i); i++){
                visited.put(i, true);
                continueStreak++;
            }

            for(int j=num-1; visited.containsKey(j); j--){
                visited.put(j, true);
                continueStreak++;
            }

            if(continueStreak > longest){
                longest = continueStreak;
            }
        }

        return longest;

    }
}

File: 128-longest-consecutive-sequence/longest-consecutive-sequence.dart
class Solution {
  int longestConsecutive(List<int> nums) {
    if(nums.isEmpty) return 0;
    Map<int, bool> map = {};

    for(int num in nums){
        map[num] = false;
    }
    int long = 0;

    for(int num in nums){
        if(map[num] == true) continue;
        map[num] = true;

        int currentStreak = 1;

        for(int i=num+1; map.containsKey(i); i++){
            map[i] = true;
            currentStreak++;
        }

        for(int j=num-1; map.containsKey(j); j--){
            map[j] = true;
            currentStreak ++;
        }

        if(currentStreak > long){
            long = currentStreak;
        }
    }
    return long;
  }
}